### PR TITLE
[MetaSchedule] Enhance Database Validation Script

### PR DESCRIPTION
**Pull request by zxybazh**
Following up on #12948, this PR further enhances the validation script by
- [x] Reusing same TIR results for speed up
- [x] Reusing RPCRunner for benchmarking & validation at the same time
- [x] Consolidate output interface and output maximum difference if wrong answer
- [x] Support Top-K validation (superfast for relay compilation validation)
- [x] Skip failed tuning records
- [x] Support Tensorized records (to be tested)
- [x] Allow batch build
- [x] Support local runner

### CODE REVIEW

**Comment by zxybazh** [comment]
Sample output looks like this:
```python
Python Environment
  TVM version    = 0.11.dev0
  Python version = 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0] (64 bit)
  os.uname()     = Linux 5.18.10-76051810-generic #202207071639~1657252310~21.10~7d5e891 SMP PREEMPT_DYNAMIC Fri J x86_64
CMake Options:
  {
    "BACKTRACE_ON_SEGFAULT": "OFF",
    "BUILD_STATIC_RUNTIME": "OFF",
    "COMPILER_RT_PATH": "3rdparty/compiler-rt",
    "CUDA_VERSION": "NOT-FOUND",
    "DLPACK_PATH": "3rdparty/dlpack/include",
    "DMLC_PATH": "3rdparty/dmlc-core/include",
    "GIT_COMMIT_HASH": "490e0e3120f304a98607770502b5700ec6ab9d55",
    "GIT_COMMIT_TIME": "2022-11-18 10:02:55 -0800",
    "HIDE_PRIVATE_SYMBOLS": "ON",
    "INDEX_DEFAULT_I64": "ON",
    "INSTALL_DEV": "OFF",
    "LLVM_VERSION": "12.0.1",
    "PICOJSON_PATH": "3rdparty/picojson",
    "RANG_PATH": "3rdparty/rang/include",
    "ROCM_PATH": "/opt/rocm",
    "SUMMARIZE": "OFF",
    "TVM_CXX_COMPILER_PATH": "/usr/bin/c++",
    "USE_ALTERNATIVE_LINKER": "AUTO",
    "USE_AOT_EXECUTOR": "ON",
    "USE_ARM_COMPUTE_LIB": "OFF",
    "USE_ARM_COMPUTE_LIB_GRAPH_EXECUTOR": "OFF",
    "USE_BLAS": "none",
    "USE_BNNS": "OFF",
    "USE_BYODT_POSIT": "OFF",
    "USE_CCACHE": "AUTO",
    "USE_CLML": "OFF",
    "USE_CLML_GRAPH_EXECUTOR": "OFF",
    "USE_CMSISNN": "OFF",
    "USE_COREML": "OFF",
    "USE_CPP_RPC": "OFF",
    "USE_CUBLAS": "OFF",
    "USE_CUDA": "/usr/local/cuda-11.8/",
    "USE_CUDNN": "ON",
    "USE_CURAND": "ON",
    "USE_CUSTOM_LOGGING": "OFF",
    "USE_CUTLASS": "OFF",
    "USE_DNNL": "OFF",
    "USE_ETHOSN": "OFF",
    "USE_FALLBACK_STL_MAP": "OFF",
    "USE_GRAPH_EXECUTOR": "ON",
    "USE_GRAPH_EXECUTOR_CUDA_GRAPH": "OFF",
    "USE_GTEST": "AUTO",
    "USE_HEXAGON": "OFF",
    "USE_HEXAGON_EXTERNAL_LIBS": "OFF",
    "USE_HEXAGON_GTEST": "/path/to/hexagon/gtest",
    "USE_HEXAGON_RPC": "OFF",
    "USE_HEXAGON_SDK": "/path/to/sdk",
    "USE_IOS_RPC": "OFF",
    "USE_KHRONOS_SPIRV": "OFF",
    "USE_LIBBACKTRACE": "ON",
    "USE_LIBTORCH": "OFF",
    "USE_LLVM": "llvm-config-12 --link-static",
    "USE_METAL": "OFF",
    "USE_MICRO": "OFF",
    "USE_MICRO_STANDALONE_RUNTIME": "OFF",
    "USE_MIOPEN": "OFF",
    "USE_MKL": "OFF",
    "USE_MSVC_MT": "OFF",
    "USE_NNPACK": "OFF",
    "USE_OPENCL": "OFF",
    "USE_OPENCL_GTEST": "/path/to/opencl/gtest",
    "USE_OPENMP": "none",
    "USE_PAPI": "OFF",
    "USE_PROFILER": "ON",
    "USE_PT_TVMDSOOP": "OFF",
    "USE_RANDOM": "ON",
    "USE_RELAY_DEBUG": "OFF",
    "USE_ROCBLAS": "OFF",
    "USE_ROCM": "OFF",
    "USE_RPC": "ON",
    "USE_RTTI": "ON",
    "USE_RUST_EXT": "OFF",
    "USE_SORT": "ON",
    "USE_SPIRV_KHR_INTEGER_DOT_PRODUCT": "OFF",
    "USE_STACKVM_RUNTIME": "OFF",
    "USE_TARGET_ONNX": "OFF",
    "USE_TENSORFLOW_PATH": "none",
    "USE_TENSORRT_CODEGEN": "OFF",
    "USE_TENSORRT_RUNTIME": "OFF",
    "USE_TFLITE": "OFF",
    "USE_TF_TVMDSOOP": "OFF",
    "USE_THREADS": "ON",
    "USE_THRUST": "OFF",
    "USE_UMA": "OFF",
    "USE_VITIS_AI": "OFF",
    "USE_VULKAN": "OFF"
  }
2022-11-21 16:19:36.788 INFO LocalBuilder: max_workers = 24
Total 1221 records to be validated. Collected in  0.000 sec.
Total 20 unique original TIR to validate. Deduplicated in  0.358 sec.
Top 10 records for each original TIR will be validated.

Progress      1 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      2 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      3 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      4 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      5 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      6 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      7 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      8 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress      9 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress     10 /    200 (estimated) checked, result:       pass, original:      0.006 ms, scheduled:      0.003 ms
Progress     11 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.030 ms
Progress     12 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.035 ms
Progress     13 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.035 ms
Progress     14 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.037 ms
Progress     15 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.038 ms
Progress     16 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.043 ms
Progress     17 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.044 ms
Progress     18 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.045 ms
Progress     19 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.047 ms
Progress     20 /    200 (estimated) checked, result:       pass, original:     52.827 ms, scheduled:      0.047 ms
Progress     21 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.104 ms
Progress     22 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.105 ms
Progress     23 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.116 ms
Progress     24 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.118 ms
Progress     25 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.122 ms
Progress     26 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.152 ms
Progress     27 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.151 ms
Progress     28 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.190 ms
Progress     29 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.218 ms
Progress     30 /    200 (estimated) checked, result:       pass, original:    146.474 ms, scheduled:      0.230 ms
Progress     31 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.013 ms
Progress     32 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.016 ms
Progress     33 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.016 ms
Progress     34 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.017 ms
Progress     35 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.018 ms
Progress     36 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.019 ms
Progress     37 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.021 ms
Progress     38 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.021 ms
Progress     39 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.024 ms
Progress     40 /    200 (estimated) checked, result:       pass, original:      7.804 ms, scheduled:      0.024 ms
Progress     41 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.086 ms
Progress     42 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.083 ms
Progress     43 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.084 ms
Progress     44 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.084 ms
Progress     45 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.084 ms
Progress     46 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.086 ms
Progress     47 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.086 ms
Progress     48 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.086 ms
Progress     49 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.087 ms
Progress     50 /    200 (estimated) checked, result:       pass, original:     43.212 ms, scheduled:      0.089 ms
Progress     51 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.057 ms
Progress     52 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.071 ms
Progress     53 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.074 ms
Progress     54 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.076 ms
Progress     55 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.082 ms
Progress     56 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.088 ms
Progress     57 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.090 ms
Progress     58 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.097 ms
Progress     59 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.098 ms
Progress     60 /    200 (estimated) checked, result:       pass, original:     65.038 ms, scheduled:      0.101 ms
Progress     61 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.074 ms
Progress     62 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.067 ms
Progress     63 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.066 ms
Progress     64 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.067 ms
Progress     65 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.075 ms
Progress     66 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.078 ms
Progress     67 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.079 ms
Progress     68 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.089 ms
Progress     69 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.096 ms
Progress     70 /    200 (estimated) checked, result:       pass, original:     31.743 ms, scheduled:      0.110 ms
Progress     71 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.006 ms
Progress     72 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.005 ms
Progress     73 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.005 ms
Progress     74 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.005 ms
Progress     75 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.006 ms
Progress     76 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.006 ms
Progress     77 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.007 ms
Progress     78 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.007 ms
Progress     79 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.008 ms
Progress     80 /    200 (estimated) checked, result:       pass, original:      3.623 ms, scheduled:      0.009 ms
Progress     81 /    196 (estimated) checked, result:       pass, original:      0.034 ms, scheduled:      0.002 ms
Progress     82 /    196 (estimated) checked, result:       pass, original:      0.034 ms, scheduled:      0.002 ms
Progress     83 /    196 (estimated) checked, result:       pass, original:      0.034 ms, scheduled:      0.002 ms
Progress     84 /    196 (estimated) checked, result:       pass, original:      0.034 ms, scheduled:      0.003 ms
Progress     85 /    196 (estimated) checked, result:       pass, original:      0.034 ms, scheduled:      0.003 ms
Progress     86 /    196 (estimated) checked, result:       pass, original:      0.034 ms, scheduled:      0.005 ms
Progress     87 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.007 ms
Progress     88 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.007 ms
Progress     89 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.009 ms
Progress     90 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.009 ms
Progress     91 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.009 ms
Progress     92 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.009 ms
Progress     93 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.010 ms
Progress     94 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.010 ms
Progress     95 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.012 ms
Progress     96 /    196 (estimated) checked, result:       pass, original:      6.110 ms, scheduled:      0.013 ms
Progress     97 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.082 ms
Progress     98 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.098 ms
Progress     99 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.101 ms
Progress    100 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.106 ms
Progress    101 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.117 ms
Progress    102 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.120 ms
Progress    103 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.125 ms
Progress    104 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.143 ms
Progress    105 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.147 ms
Progress    106 /    196 (estimated) checked, result:       pass, original:     43.432 ms, scheduled:      0.150 ms
Progress    107 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.051 ms
Progress    108 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.048 ms
Progress    109 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.051 ms
Progress    110 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.053 ms
Progress    111 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.061 ms
Progress    112 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.068 ms
Progress    113 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.082 ms
Progress    114 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.086 ms
Progress    115 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.078 ms
Progress    116 /    196 (estimated) checked, result:       pass, original:     24.410 ms, scheduled:      0.102 ms
Progress    117 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.067 ms
Progress    118 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.081 ms
Progress    119 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.095 ms
Progress    120 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.112 ms
Progress    121 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.113 ms
Progress    122 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.125 ms
Progress    123 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.129 ms
Progress    124 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.130 ms
Progress    125 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.142 ms
Progress    126 /    196 (estimated) checked, result:       pass, original:     32.747 ms, scheduled:      0.142 ms
Progress    127 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.051 ms
Progress    128 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.059 ms
Progress    129 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.059 ms
Progress    130 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.059 ms
Progress    131 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.060 ms
Progress    132 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.060 ms
Progress    133 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.063 ms
Progress    134 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.065 ms
Progress    135 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.071 ms
Progress    136 /    196 (estimated) checked, result:       pass, original:     31.283 ms, scheduled:      0.077 ms
Progress    137 /    191 (estimated) checked, result:       pass, original:      0.000 ms, scheduled:      0.002 ms
Progress    138 /    191 (estimated) checked, result:       pass, original:      0.000 ms, scheduled:      0.002 ms
Progress    139 /    191 (estimated) checked, result:       pass, original:      0.000 ms, scheduled:      0.002 ms
Progress    140 /    191 (estimated) checked, result:       pass, original:      0.000 ms, scheduled:      0.002 ms
Progress    141 /    191 (estimated) checked, result:       pass, original:      0.000 ms, scheduled:      0.002 ms
Progress    142 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.010 ms
Progress    143 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.016 ms
Progress    144 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.021 ms
Progress    145 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.025 ms
Progress    146 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.029 ms
Progress    147 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.030 ms
Progress    148 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.030 ms
Progress    149 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.033 ms
Progress    150 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.033 ms
Progress    151 /    191 (estimated) checked, result:       pass, original:      0.338 ms, scheduled:      0.033 ms
Progress    152 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    153 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    154 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    155 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    156 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    157 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    158 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    159 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    160 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    161 /    191 (estimated) checked, result:       pass, original:      0.177 ms, scheduled:      0.005 ms
Progress    162 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.049 ms
Progress    163 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.048 ms
Progress    164 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.053 ms
Progress    165 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.055 ms
Progress    166 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.056 ms
Progress    167 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.056 ms
Progress    168 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.057 ms
Progress    169 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.059 ms
Progress    170 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.062 ms
Progress    171 /    191 (estimated) checked, result:       pass, original:     70.768 ms, scheduled:      0.064 ms
Progress    172 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.058 ms
Progress    173 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.055 ms
Progress    174 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.059 ms
Progress    175 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.063 ms
Progress    176 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.063 ms
Progress    177 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.072 ms
Progress    178 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.074 ms
Progress    179 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.080 ms
Progress    180 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.074 ms
Progress    181 /    191 (estimated) checked, result:       pass, original:     31.186 ms, scheduled:      0.085 ms
Progress    182 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.048 ms
Progress    183 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.044 ms
Progress    184 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.050 ms
Progress    185 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.052 ms
Progress    186 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.065 ms
Progress    187 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.067 ms
Progress    188 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.068 ms
Progress    189 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.077 ms
Progress    190 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.078 ms
Progress    191 /    191 (estimated) checked, result:       pass, original:     24.627 ms, scheduled:      0.082 ms
Validation finished! Total time spent:  168.474 sec.
```

File: python/tvm/meta_schedule/testing/validate_database.py [hunk 1]
```diff
@@ -124,158 +252,495 @@ def default_input_generator(mod: IRModule) -> List[tvm.nd.NDArray]:
...
+                        inputs,
+                        original_mod,
+                        scheduled_mod,
+                        str(record.trace),
+                        original_res=original_res,
+                        original_run_secs=original_run_secs,
+                        print_result=print_result,
+                    )
+                    # create rpc runner
+                    runner = ms.runner.RPCRunner(
```
Comment (yelite) [inline python/tvm/meta_schedule/testing/validate_database.py:717]: Will it be useful to make rpc optional, which means it uses LocalRunner if no rpc related args are provided?

Comment (zxybazh) [reply python/tvm/meta_schedule/testing/validate_database.py:717]: I think that's a reasonable ask, what do you think would be a good user interface?
Say, `--rpc-host="local"`?

Comment (yelite) [reply python/tvm/meta_schedule/testing/validate_database.py:717]: Maybe just make the `--rpc-host` arg and its friends optional and use local runner if `--rpc-host` is not given by user

Comment (zxybazh) [reply python/tvm/meta_schedule/testing/validate_database.py:717]: Sure, can do that.

Comment (zxybazh) [reply python/tvm/meta_schedule/testing/validate_database.py:717]: I'll throw a warning though, to make sure user is aware of that.

**Review by yelite — APPROVED** [review]

**Pull request merged**